api/envoy/service/discovery/v2/csds.proto

```diff
@@ -0,0 +1,85 @@
+syntax = "proto3";
+
+package envoy.service.discovery.v2;
+
+option java_package = "io.envoyproxy.envoy.service.discovery.v2";
+option java_outer_classname = "CsdsProto";
+option java_multiple_files = true;
+option java_generic_services = true;
+
+import "envoy/admin/v2alpha/config_dump.proto";
+import "envoy/api/v2/core/base.proto";
+
+import "google/api/annotations.proto";
+import "google/protobuf/struct.proto";
+
+// CSDS is Client Status Discovery Service. It can be used to get the status of
+// xDS-compliant client from the management server's point of view.
```

Comment:
When I had originally skimmed this PR, one of the things that I thought this might cover is a more systematic way of fetching the current state *from the client*. So basically a formalized version of config dump. This would allow a "client tracking server" to potentially get updates from Envoy, etc. on its current status to help with reconciliation. Is this something you have thought about? It's not the same as what is being proposed here but is closely related IMO.

Reply:
Thanks! Yes, I have thought about that and had some discussion with htuch@. If it's just configs, we could re-use this API to fetch configs from Envoy as well. However, if we also want to expose stats, then the stats from control plane and from Envoy can be quite different, and it would be confusing to use the same API.
Alternatively, we could have another API that's basically Admin Discovery Service, where Envoy can expose it's admin interface (configs, stats).  WDYT?

Reply:
@htuch for further inputs.

Reply:
I think there are a bunch of ways we could go here per your comment. I'm CCing @sindhura who is working on a system at Lyft which right now going to fetch most data directly from the admin server. But I think it would be great if there were a more formalized way of accomplishing this. Hopefully she can provide some thoughts (probably delayed due to holidays).

Reply:
I think it really depends on what we want to be fetching. I think some parts of the admin interface are sort of covered by the idea of a `ConfigDump` style API. This could be extended to capture something closer to the dynamic aspect of `/clusters` (e.g. with health status).

Sometime stats comes up in these conversations, I think stats is really something that shouldn't be handled at all by some new API; we have an existing extension point and ways of moving stats out, the admin endpoint here is just a convenience.

Reply:
I have updated the comments to mention that this can also be used as an interface to fetch client state. PTAL. Thanks!

Reply:
Updated. PTAL.